### ISSUE TITLE
No images rendering in Windows Server with IIS

### ISSUE THREAD

**Issue by Avrgebro** [dompdf/dompdf#2908]
Lately I've developed a Laravel application that generates a PDF, the development process was made in ubuntu 21.10 using laravel sail, and it has been deployed to a Windows Server with IIS 10.

The PDF contains an image that is rendered using an encoded svg:

`<img style="width: 200px;" src="data:image/svg+xml;base64,{{ $base64_image }}" alt="">`

the `$base64_image` is obtained by the following line of code:

`base64_encode(str_replace('<svg', '<svg fill="#000000" ',file_get_contents("img/image.svg"))),`

this line uses the `file_get_contents()` function to retrieve the svg image, then it fills the svg with #000000 (black), and finally converts to a base64 string.

So basically what my `<img>` receives is a plain string. This approach works flawlessly in my dev server with ubuntu, but in IIS no image is showing at all.

What I have tried so far:

Updating MIME Types in the IIS site manager.
 - Updating the web.config to allow SVG
 - I've used an url for a PNG image hosted online
 - Discarded SSL issues
 - Hardcoded the base64 string in the  `<img>`  tag `src="data:image/svg+xml;base64,P5Gry.....Or2="`

Basically on the IIS server no image is showing at all when generating the PDF even when hardcoding the base64 encoded png or svg string, just a square with the message.

> image not found or type unknown

Is this something that can be solved? it works great in every ubuntu server i've tested.

Thanks!

Edit:

Im using `"dompdf/dompdf": "^1.2"`

**Comment by bsweeney** [comment]
One known issue (though I haven't checked if it's still a problem) is that white space in base64 encoded content breaks things (#1386). You might check your base64 image to see if that's possibly what's going on.

You might want to enable some debug logging so you can get some messaging on why that's failing.

```
ini_set("display_errors", true);
ini_set("error_log", "phperr.log");
ini_set("log_errors", true);
error_reporting(E_ALL);

$_dompdf_show_warnings = true; // 
$dompdf = new Dompdf([
    "logOutputFile" => "log.html",
    "debugPng" => false, // extra messaging
    "debugKeepTemp" => false, // don't delete temp files
]);

// do things
```

**Comment by Avrgebro** [comment]
Thanks for your answer!!
I followed your instructions and got this in my log.html:

```
<span style='color: #000' title='Frames'>  1481</span>
<span style='color: #009' title='Memory'>  47104.00 KB</span>
<span style='color: #900' title='Time'>    730.64 ms</span>
<span  title='Quirksmode'>
	<span style='color: #0d0'>OFF</span>
</span><br />
```

I tried a different approach, instead of using `<img>` I used `<div>` and set a width, height and bg image:

```
.logo {
    background-image: url('https://www.domain.com/path/to/image/main-logo.png');
    background-position: center;
    background-size: contain;
    background-repeat: no-repeat;
    height: 50px;
    width: 200px;
}
```

It worked perfectly locally and in the dev server, how ever when deploying to IIS, i just got an empty space, this same image loads fine anywhere else in the webpage but does not load in the rendered pdf.

**Comment by Avrgebro** [comment]
I was able to solve this problem.

My previous comment made me think that rather than being an issue with DomPDF, it was an issue with IIS blocking the content somehow. after some investigation I found: https://stackoverflow.com/questions/10512053/css-images-js-not-loading-in-iis

Basically IIS blocks static content by default, and a PDF being a static document, was not able to render properly, allowing static content in the side solved the issue.

Sorry for creating an issue unrelated to DomPDF, but hopefully if someone else has this problem in the future, they can find this and solve it, instead of banging their head against the wall for two days like i did.
